lang/java/compiler/src/main/java/org/apache/avro/compiler/idl/ResolvingVisitor.java

```diff
@@ -139,10 +139,7 @@ public SchemaVisitorAction afterVisitNonTerminal(final Schema nt) {
         List<Schema.Field> fields = nt.getFields();
         List<Schema.Field> newFields = new ArrayList<>(fields.size());
         for (Schema.Field field : fields) {
-          Schema.Field newField = new Schema.Field(field.name(), replace.get(field.schema()), field.doc(),
-              field.defaultVal(), field.order());
-          copyAllProperties(field, newField);
-          newFields.add(newField);
+          newFields.add(new Field(field, replace.get(field.schema())));
```

Comment:
In the previous version, `field.defaultVal()` results in `null` for **all** default values if the type is not yet resolved.
By using the `Field(Field, Schema)` constructor, the underlying JSON object is used, preserving the default value.